tests/Resources/Page/ODH/Prometheus/Prometheus.robot

```diff
@@ -0,0 +1,46 @@
+*** Settings ***
+Library        Collections
+Library        Process
+Library        RequestsLibrary
+
+*** Keywords ***
+Run Query
+  [Arguments]  ${pm_url}  ${pm_token}  ${pm_query}
+  ${pm_headers}=       Create Dictionary  Authorization=Bearer ${pm_token}
+  ${resp}=       GET   url=${pm_url}/api/v1/query?query=${pm_query}   headers=${pm_headers}   verify=${False}
+  Status Should Be    200    ${resp}
+  Log To Console    ${resp.json()}
+  [Return]   ${resp}
+
+Get Rules
+  ${pm_headers}=       Create Dictionary  Authorization=Bearer ${pm_token}
+  [Arguments]  ${pm_url}  ${pm_token}  ${rule_type}
+  ${resp}=       GET   url=${pm_url}/api/v1/rules?type=${rule_type}   headers=${pm_headers}   verify=${False}
+  Status Should Be    200    ${resp}
+  [Return]  ${resp.json()}
+
+Verify Rule
+  [Arguments]  ${rule_group}  @{all_rules}
+
+  FOR    ${rule}    IN    @{all_rules}
+    ${rule_name}=  Get From Dictionary  ${rule}  name
+    ${rules_list}=  Get From Dictionary  ${rule}  rules
+
+    IF  '${rule_name}' == '${rule_group}'
+      ${rules_list_len}=  Get Length  ${rules_list}
+      Should Be True  ${rules_list_len} != 0
+      Return from keyword  ${TRUE}
+    END
+  END
+  Log To Console  ${rule_group} was not found in Prometheus rules
+  Fail
+
+Verify Rules
```

Comment:
This keyword could be renamed to:  `Rules Should Contain` (following the SeleniumLibrary naming) with the same parameters or create different keywords for each type:

```
Rules Should Contain Alert
Rules Should Contain Record
```